pkg/agent/agent.go

```diff
@@ -111,6 +112,9 @@ type Agent struct {
 	// Cloud API.
 	metricsBackoffUntil time.Time
 
+	// Used to accumulate metrics for a same timestamp before pushing them to the cloud.
+	metricStack map[string][]*io_prometheus_client.MetricFamily
```

Comment:
bike shed: `metricStack` doesn't quite capture the variables purpose. Can we maybe name it something like `aggregatedMetrics` or `queuedMetrics`?  Other ideas?